substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrChunkWriter.java

```diff
@@ -57,27 +61,32 @@
  */
 public final class JfrChunkWriter implements JfrUnlockedChunkWriter {
     private static final byte[] FILE_MAGIC = {'F', 'L', 'R', '\0'};
-    private static final int JFR_VERSION_MAJOR = 2;
-    private static final int JFR_VERSION_MINOR = 0;
+    private static final short JFR_VERSION_MAJOR = 2;
+    private static final short JFR_VERSION_MINOR = 0;
```

Comment:
Yes. My understanding from reading the source of 11/latest is that 2.1 introduced to help support jfr streaming (I think) and includes a chunk header checkpoint at the end of every chunk. There is an extra `metadata.xml` entry for type `ChunkHeader` that isn't in 11. You can see more about it here:
https://github.com/openjdk/jdk/blob/master/src/hotspot/share/jfr/recorder/repository/jfrChunkWriter.cpp#L192

In our PR we tried to implement a specific commit in latest instead of in 11 in order to best support streaming and future features so we were working to also have this chunk header checkpoint. I got stuck on this `metadata.xml` being different between 11 and latest, so was looking at having a substratevm specific `metadata.xml` to decouple ourselves from the jdk implementation.

I think in the long term I'd like to have a decoupled `metadata.xml` to also reduce the size of the metadata and constant pool events. (e.g. metadata event is currently 617544 bytes for test code that emits a single custom event, when it really doesn't need to be). But 1. It's on the backburner in terms of priorty and 2. I'm not strongly for/against anything here. It's a tradeoff to think about.

Reply:
Ah; I mean to have a SubstrateVM specific metadata (xml file or maybe created during the build process) that isn't a carbon copy from the JDK that is being used to build the native image. My hope is to be able to add/remove metadata as we see fit. One to potentially support jfr v2.1 with the `ChunkHeader` type and two to drop all of the HotSpot specific metadata like Shenandoah/Z1 events, etc.